x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/action/filter/SecurityActionFilter.java

```diff
@@ -87,38 +84,25 @@ public SecurityActionFilter(AuthenticationService authcService, AuthorizationSer
             throw LicenseUtils.newComplianceException(XPackField.SECURITY);
         }
 
-        if (licenseState.isSecurityEnabled()) {
-            final ActionListener<Response> contextPreservingListener =
-                    ContextPreservingActionListener.wrapPreservingContext(listener, threadContext);
-            final boolean useSystemUser = AuthorizationUtils.shouldReplaceUserWithSystem(threadContext, action);
-            try {
-                if (useSystemUser) {
-                    securityContext.executeAsUser(SystemUser.INSTANCE, (original) -> {
-                        applyInternal(task, chain, action, request, contextPreservingListener);
-                    }, Version.CURRENT);
-                } else if (AuthorizationUtils.shouldSetUserBasedOnActionOrigin(threadContext)) {
-                    AuthorizationUtils.switchUserBasedOnActionOriginAndExecute(threadContext, securityContext, (original) -> {
-                        applyInternal(task, chain, action, request, contextPreservingListener);
-                    });
-                } else {
-                    try (ThreadContext.StoredContext ignore = threadContext.newStoredContext(true)) {
-                        applyInternal(task, chain, action, request, contextPreservingListener);
-                    }
-                }
-            } catch (Exception e) {
-                listener.onFailure(e);
-            }
-        } else if (SECURITY_ACTION_MATCHER.test(action)) {
-            if (licenseState.isSecurityEnabled() == false) {
-                listener.onFailure(new ElasticsearchException("Security must be explicitly enabled when using a [" +
-                        licenseState.getOperationMode().description() + "] license. " +
```

Comment:
I think it would be good to still return an error when security is disabled and one of the Security APIs is invoked.
It's more of an core ask than a Security ask, so I'm content with opening an issue for it, if you also think this is still a good idea, ofc.

Reply:
> It's more of an core ask than a Security ask, so I'm content with opening an issue for it, 

SGTM!